services/queue/src/artifacts.js

```diff
@@ -449,29 +484,16 @@ builder.declare({
     '',
     '* location: the url of the artifact if a redirect is to be performed',
     '* x-taskcluster-artifact-storage-type: the storage type.  Example: s3',
-    '',
-    'The following important headers are set on responses to this method for Blob artifacts',
-    '',
-    '* x-taskcluster-location-content-sha256: the SHA256 of the artifact',
-    '*after* any content-encoding is undone.  Sha256 is hex encoded (e.g. [0-9A-Fa-f]{64})',
-    '* x-taskcluster-location-content-length: the number of bytes *after* any content-encoding',
-    'is undone',
-    '* x-taskcluster-location-transfer-sha256: the SHA256 of the artifact',
-    '*before* any content-encoding is undone.  This is the SHA256 of what is sent over',
-    'the wire.  Sha256 is hex encoded (e.g. [0-9A-Fa-f]{64})',
-    '* x-taskcluster-location-transfer-length: the number of bytes *after* any content-encoding',
-    'is undone',
-    '* x-taskcluster-location-content-encoding: the content-encoding used.  It will either',
-    'be `gzip` or `identity` right now.  This is hardcoded to a value set when the artifact',
-    'was created and no content-negotiation occurs',
-    '* x-taskcluster-location-content-type: the content-type of the artifact',
   ].join('\n'),
 }, async function(req, res) {
   let taskId = req.params.taskId;
   let runId = parseInt(req.params.runId, 10);
   let name = req.params.name;
+  let names = [name];
 
-  return replyWithArtifact.call(this, taskId, runId, name, req, res);
+  await req.authorize({ names });
+
+  return replyWithArtifact.call(this, taskId, runId, name, req, res, names);
```

Comment:
It seems like misplaced responsibilityfor the caller to remember to initialize `names = [name]`, and call `req.authorize({ names })`. If this was Python, I'd make the function signature `names = None`, and do this work in ``replyWithArtifact``. I'm not sure if JavaScript has a similar pattern. 

But, it also appears to be called from just two places in the same file, so maybe OK?

Reply:
I agree, this is a little ugly.  I had actually used default arguments in an earlier iteration, but switched to being explicit because the same value is used in the `req.authorize` above.  And the `req.authorize` needs to happen early to avoid a new instance of #4215.  Anyway, this is all internal to the file, so I agree with "maybe OK" :)